install_script_opw.sh

```diff
@@ -0,0 +1,644 @@
+#!/bin/bash
+# (C) Datadog, Inc. 2010-present
+# All rights reserved
+# Licensed under Apache-2.0 License (see LICENSE)
+# Datadog Observability Pipelines Worker installation script:
+# install and set up the Observability Pipelines Worker on supported Linux distributions
+# using the package manager and Datadog repositories.
+
+set -e
+
+install_script_version="0.1.0"
+logfile="dd-install.log"
+support_email="support@datadoghq.com"
+
+# DATADOG_APT_KEY_CURRENT.public always contains key used to sign current
+# repodata and newly released packages
+# DATADOG_APT_KEY_382E94DE.public expires in 2022
+# DATADOG_APT_KEY_F14F620E.public expires in 2032
+# DATADOG_APT_KEY_C0962C7D.public expires in 2028
+APT_GPG_KEYS=("DATADOG_APT_KEY_CURRENT.public" "DATADOG_APT_KEY_C0962C7D.public" "DATADOG_APT_KEY_F14F620E.public" "DATADOG_APT_KEY_382E94DE.public")
+
+# DATADOG_RPM_KEY_CURRENT.public always contains key used to sign current
+# repodata and newly released packages
+# DATADOG_RPM_KEY_E09422B3.public expires in 2022
+# DATADOG_RPM_KEY_FD4BF915.public expires in 2024
+# DATADOG_RPM_KEY_B01082D3.public expires in 2028
+RPM_GPG_KEYS=("DATADOG_RPM_KEY_CURRENT.public" "DATADOG_RPM_KEY_B01082D3.public" "DATADOG_RPM_KEY_FD4BF915.public" "DATADOG_RPM_KEY_E09422B3.public")
+
+# Error codes for telemetry
+GENERAL_ERROR_CODE=1
+UNSUPPORTED_PLATFORM_CODE=5
+INVALID_PARAMETERS_CODE=6
+
+# Set up a named pipe for logging
+npipe=/tmp/$$.tmp
+mknod $npipe p
+
+# Log all output to a log for error checking
+tee <$npipe $logfile &
+exec 1>&-
+exec 1>$npipe 2>&1
+trap 'rm -f $npipe' EXIT
+
+function fallback_msg(){
+  printf "
+If you are still having problems, please send an email to $support_email
+with the contents of $logfile and any information you think would be
+useful and we will do our very best to help you solve your problem.\n"
+}
+
+# TODO: this uses an Agent specific endpoint, remove before using?
+function report(){
+  if curl -f -s \
+    --data-urlencode "os=${OS}" \
+    --data-urlencode "version=${worker_major_version}" \
+    --data-urlencode "log=$(cat $logfile)" \
+    --data-urlencode "email=${email}" \
+    --data-urlencode "apikey=${apikey}" \
+    "$report_failure_url"; then
+   printf "A notification has been sent to Datadog with the contents of $logfile\n"
+  else
+    printf "Unable to send the notification (curl v7.18 or newer is required)"
+  fi
+}
+
+function report_telemetry() {
+  if [ "$DD_INSTRUMENTATION_TELEMETRY_ENABLED" == "false" ] || \
+    [ "$site" == "ddog-gov.com" ] || \
+    [ -z "${apikey}" ] || \
+    [ -z "$telemetry_url" ]; then
+    return
+  fi
+
+  if [ -n "$worker_minor_version" ] ; then
+    safe_worker_version=$(echo -n "$worker_major_version.$worker_minor_version" | tr '\n' ' ' | tr '"' '_')
+  else
+    safe_worker_version=$(echo -n "$worker_major_version" | tr '\n' ' ' | tr '"' '_')
+  fi
+
+  if [ -z "${ERROR_CODE}" ] ; then
+    telemetry_event="
+{
+   \"request_type\": \"apm-onboarding-event\",
```

Comment:
Should this be something specific to OP?  This could be really helpful telemetry data 😄

Reply:
I actually have no idea where this goes, it sounds like a generic event that gets filtered down by the `event_name` and the tags - it would be good to know where this gets surfaced in the UI.

Reply:
I dug about the UI some and didn't find it, I expect we can get some more info next week from the platform team 👍

Reply:
@sghall - discussed with @bkabrda, and unfortunately this is unlikely to make the cut for the first version. We _do_ want to include it, and will make sure it gets back in as soon as the backend can handle the payloads properly.